pandas/core/series.py

```diff
@@ -2080,10 +2080,24 @@ def __rmatmul__(self, other):
     @Appender(base._shared_docs['searchsorted'])
     @deprecate_kwarg(old_arg_name='v', new_arg_name='value')
     def searchsorted(self, value, side='left', sorter=None):
-        if sorter is not None:
-            sorter = ensure_platform_int(sorter)
-        return self._values.searchsorted(Series(value)._values,
-                                         side=side, sorter=sorter)
+        simple_types = (is_integer_dtype, is_float_dtype, is_object_dtype,
+                        is_categorical_dtype)
+
+        if any(is_dtype(self) for is_dtype in simple_types):
+            result = com.searchsorted(self._values, value,
+                                      side=side, sorter=sorter)
+        else:
+            # e.g. self is datetimelike and value is a pd.Timestamp
+            if sorter is not None:
+                sorter = ensure_platform_int(sorter)
+            value = Series(value)._values
+            result = self._values.searchsorted(value, side=side, sorter=sorter)
+
+        if is_scalar(result):
+            # ensure that a 1-dim array is returned
+            result = np.array([result])
```

Comment:
This bit is just to maintain backward compatability, to make <series.searchsorted always return a 1-dim array. I'd like to remove this in the future.